draft-ietf-ppm-dap.md

```diff
@@ -1214,64 +1269,174 @@ as follows:
+request is an `AggregationJob` (see {{aggregation-job-representation}}).
+`AggregationJob.round` MUST be set to the Leader's current round (i.e., one more
+than the `round` value in the Helper's latest `AggregationJob` message) and the
+`prepare_steps` field MUST be a sequence of `PrepareStep`s in the `continued`
+state containing the corresponding `inbound` prepare message.
+
+###### Helper Continuation
+
+If the Helper does not recognize the task ID, then it MUST abort with error
+`unrecognizedTask`. If the Helper does not recognize the aggregation job, then
+it MUST abort with error `unrecognizedAggregationJob`. This could occur because
+the aggregation job never existed, but also if it was deleted (see
+{{aggregation-job-delete}}).
+
+Otherwise, for each `PrepareStep` received from the Leader in
+`AggregationJob.prepare_steps`, the Helper checks
+`PrepareStep.prepare_step_state` to determine if the report share should
+continue being prepared. If the report share is in state `failed` or `finished`,
+then the Helper marks the report as failed with error `unrecognized_message`.
+
+If the state is `continued`, then the Helper continues with preparation for the
+report share. The Helper MUST check its current preparation state against the
+Leader's `AggregationJob.round` value. If the Leader is one round ahead of the
+Helper, then the Helper can combine the Leader's prepare message and the
+Helper's current preparation state (`prep_state`) as follows:
 
-struct {
-  PrepareStep prepare_steps<1..2^32-1>;
-} AggregateInitializeResp;
 ~~~
+out = VDAF.prep_next(prep_state, inbound)
+~~~
+
+where `inbound` is the previous VDAF prepare message sent by the Leader and
+`prep_state` is the Helper's current preparation state. This step yields one of
+three outputs:
+
+1. An error, in which case the input report share is marked as failed and the
+   Helper replies to the Leader with a `PrepareStep` in the `failed` state.
+1. An output share, in which case the Helper stores the output share for future
+   collection as described in {{collect-flow}} and replies to the Leader with a
+   `PrepareStep` in the `finished` state. The Helper expects no further requests
+   from the Leader for a report in the `finished` state.
+1. An updated VDAF state and preparation message, denoted
+   `(prep_state, prep_msg)`, in which case the Helper replies to the leader with
+   `PrepareStep` in the `continued` state containing `prep_msg`.
 
-The message is a sequence of PrepareStep values, the order of which matches that
-of the ReportShare values in `AggregateInitializeReq.report_shares`. Each report
-that was marked as invalid is assigned the PrepareStepResult `failed`.
-Otherwise, the PrepareStep is either marked as continued with the output
-`prep_msg`, or is marked as finished if the VDAF preparation process is finished
-for the report share.
+After computing each report share's preparation state and prepare message, the
```

Comment:
Recovering from round skew only requires storing one old round of responses; I think it would be worthwhile to mention this.

Reply:
- [ ] include implementation note explaining that helper only needs to store the most recent set of prepare messages for each aggregation job